app/services/payment_provider_customers/cashfree_service.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module PaymentProviderCustomers
+  class CashfreeService < BaseService
+    include Customers::PaymentProviderFinder
+
+    def initialize(cashfree_customer = nil)
+      @cashfree_customer = cashfree_customer
+
+      super(nil)
+    end
+
+    def create
+      result.cashfree_customer = cashfree_customer
+      result
+    end
+
+    def update
+      result
+    end
+
+    def generate_checkout_url(send_webhook: true)
+      return result.not_found_failure!(resource: 'cashfree_payment_provider') unless cashfree_payment_provider
+      result.service_failure!(code: 'generate_checkout_url_error', message: 'Cashfree payment provider does not support the checkout url feature')
```

Comment:
Since the feature is not supported, maybe you could just return an empty result, rather than a `service_failure`, what do you think?

Reply:
Yes, we can return an empty result, but how will the user know that the feature is not supported? The user might consider it a bug.

Reply:
Yes I see. It will be exposed only in the APIs

`service_failre` is not supposed to be returned to the end users as it does not have a proper formatter in both REST and GraphQL APIs.

You could use `result.not_allowed_failure(code: 'feature_not_supported')` that way, the APIs will render a proper `HTTP 405` error.